sw/device/lib/testing/spi_flash_emulator.c

```diff
@@ -101,7 +107,15 @@ static status_t prepare_sfdp(dif_spi_host_t *spih,
 
   TRY(dif_spi_device_write_flash_buffer(spid, kDifSpiDeviceFlashBufferTypeSfdp,
                                         0, sizeof(sfdp), (uint8_t *)&sfdp));
-  return OK_STATUS();
+
+  uint32_t quad_enable = 0;
+  if (length >= 14) {
+    // JESD216F, section 6.4.18:
+    // The Quad Enable mechanism is bits 20:23 of the 15th dword.
+    quad_enable = read_32(data + offset + 14 * sizeof(uint32_t));
+    quad_enable = bitfield_field32_read(quad_enable, QUAD_ENABLE);
+  }
+  return OK_STATUS(quad_enable);
```

Comment:
Left a TODO comment to restructure.  There are other SFDP properties the emulator _should_ care about, such as the block erase opcode types.  Since I don't yet have the capability to drive dual, quad or any alternative opcodes from the client, I don't think its worthwhile to try to fix this just yet.